packages/hooks/src/utils/useBaseQuery.ts

```diff
@@ -18,21 +18,39 @@ export function useBaseQuery<TData = any, TVariables = OperationVariables>(
   const context = useContext(getApolloContext());
   const [tick, forceUpdate] = useReducer(x => x + 1, 0);
   const updatedOptions = options ? { ...options, query } : { query };
+  const isRendering = useRef(true);
```

Comment:
Why not just have a useState hook for each then if you really want have separate useEffect to subscribe to any variable changes? Just curious. I mean if it works it works I just don't understand why the useRef versus useState.

Reply:
In this case I'm interested in using `isRendering` and `isRenderScheduled` to track render metadata, not to trigger an actual render. Doing this is easier with `useRef` since we don't ever have to worry about triggering an unintended render. We could likely achieve the same thing with `useState` and guarding/effect dep tracking, but in all honesty I kinda like refs 🙂.

Reply:
@hwillson wow that's a clever trick. I would have just used ```useState``` but you're right in that ```useRef``` won't trigger an unintended render and it's mutable. That's pretty slick. I'll have to try this out sometime. From the React docs: However, useRef() is useful for more than the ref attribute. It’s handy for keeping any mutable value around similar to how you’d use instance fields in classes... useRef will give you the same ref object on every render.